packages/next/src/lib/url.ts

```diff
@@ -21,14 +21,17 @@ export function stripNextRscUnionQuery(relativeUrl: string): string {
   return urlInstance.pathname + urlInstance.search
 }
 
-export function canParseUrl(url: string): boolean {
-  if (typeof URL.canParse === 'function') {
-    return URL.canParse(url)
+// patch URL.canParse to be available in older browsers e.g. Safari 16
+// x-ref: https://github.com/vercel/next.js/pull/70215
+// x-ref: https://caniuse.com/?search=URL.canParse
+// Since URL.canParse is available in Node.js v18.17.0 and later,
+// we only need to patch it for the browser.
+if (typeof window !== 'undefined' && !('canParse' in URL)) {
```

Comment:
Need to add the URL.canParse polyfill in `@next/polyfill-nomodule` where it gets applied globally

Reply:
TIL